targets/TARGET_Samsung/TARGET_SIDK_S5JS100/device/TOOLCHAIN_IAR/sidk_s5js100.icf

```diff
@@ -0,0 +1,55 @@
+/****************************************************************************
+ *
+ * Copyright 2019 Samsung Electronics All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+
+/* Linker file for the IAR Compiler for ARM */
+/* Specials */
+/* Meory Regions */
+define symbol S5JS100_BOOTMEM_BASE     = 0x00000000;
+define symbol S5JS100_BOOTMEM_END     = S5JS100_BOOTMEM_BASE + 8K;
+define symbol S5JS100_IRAM_BASE     = 0x00100000;
+define symbol S5JS100_IRAM_END      = S5JS100_IRAM_BASE + 512K;
+define symbol S5JS100_CODE_BASE     = 0x406F4000;
+define symbol S5JS100_CODE_END      = S5JS100_CODE_BASE + 1492K;
+define symbol S5JS100_FLASH_BASE    = 0x40000000;
+
+/* Stack Size & Heap Size*/
+if (!isdefinedsymbol(MBED_BOOT_STACK_SIZE)) {
+    define symbol MBED_BOOT_STACK_SIZE = 0x400;
+}
+
+define symbol CSTACK_SIZE   = MBED_BOOT_STACK_SIZE;
+define symbol HEAP_SIZE     = 0x50000;
```

Comment:
You define it as `__heap_size__ = 0x4000;` for GCC, also why not use dynamic heap sizing based on available space?

Reply:
@bulislaw 
Thanks for the comment.
Could you kindly show me the example of other target driver doing the dynamic heap sizing or instruction on how to do so? I searched and found this which is similar but I am not sure if you are referring to this.
```
HEAP_SIZE  = DEFINED(__heap_size__)  ? __heap_size__  : 0x0C00;
```

@jh6186,
The heap size of IAR (0x50000) and GCC (0x4000) seem to be different. Is this expected?
```
TARGET_SIDK_S5JS100/device/TOOLCHAIN_IAR/sidk_s5js100.icf:36:define symbol HEAP_SIZE     = 0x50000;
TARGET_SIDK_S5JS100/device/TOOLCHAIN_IAR/sidk_s5js100.icf:45:define block HEAP   with alignment = 8, size = HEAP_SIZE   { };
TARGET_SIDK_S5JS100/device/TOOLCHAIN_GCC_ARM/sidk_s5js100.ld:78:__heap_size__ = 0x4000;
TARGET_SIDK_S5JS100/device/TOOLCHAIN_GCC_ARM/sidk_s5js100.ld:80:HEAP_SIZE  = DEFINED(__heap_size__)  ? __heap_size__  : 0x0400;
```

Reply:
@andrewc-arm 
For GCC, the HEAP_SIZE never referred, please remove it.
For IAR, leave 0x50000 as static size

Reply:
IAR supports dynamic, please define `define block HEAP       with expanding size, alignment = 8, minimum size = __ICFEDIT_size_heap__    { };` - and set `__ICFEDIT_size_heap__  ` to minimum size

Reply:
Thank you, @jh6186, @0xc0170 and @bulislaw 
Updated with 374aed24111d80b03efc096722daa84234cfb613. And will run whole Greentea test to make sure it's safe.